opm/parser/eclipse/Parser/MessageContainer.cpp

```diff
@@ -25,94 +25,127 @@
 namespace Opm {
 
 
-    void MessageContainer::error(const std::string& msg, const std::string& filename, const int lineno)
-    {
-        std::unique_ptr<Location> loc(new Location{filename, lineno});
-        m_messages.emplace_back(Message{MessageType::MessageTypeEnum::Error, msg, std::move(loc)});
+    void MessageContainer::error( const std::string& msg,
+                                  const std::string& filename,
+                                  const int lineno ) {
+        m_messages.emplace_back( Message {
```

Comment:
The compiler is unable to deduce it correctly without the hint from `Message {}` (or constructor definitions). I use `emplace_back` just to emphasise the move going on.

An alternative is surely to define two constructors for `Message`, in which case the `Message {}` can go away.